src/components/SingletonEditor.tsx

```diff
@@ -41,6 +65,7 @@ export const SingletonEditorProvider = ({
   children: React.ReactNode;
 }) => {
   const [content, setContent] = useState<Content | undefined>();
+  const onSave = useRef(() => undefined);
```

Comment:
Very minor, it is the same, but I think that no return (or return _nothig/void_) is semantically more correct than returning _undefined_.

```suggestion
  const onSave = useRef(() => {});
```